contracts/exchangeIssuance/ExchangeIssuanceLeveraged.sol

```diff
@@ -0,0 +1,621 @@
+     */
+    function executeOperation(
+        address[] calldata assets,
+        uint256[] calldata amounts,
+        uint256[] calldata premiums,
+        address , // 
+        bytes calldata params
+    ) external override returns (bool) {
+        require(assets.length == 1, "Exchange Issuance Leveraged: TOO MANY ASSETS");
+        require(amounts.length == 1, "Exchange Issuance Leveraged: TOO MANY AMOUNTS");
+        require(premiums.length == 1, "Exchange Issuance Leveraged: TOO MANY PREMIUMS");
+
+        _depositLongToken(assets[0], amounts[0]);
+        _issueSet(params);
+        _obtainLongTokens(assets[0], amounts[0] + premiums[0], params);
+        _approveAssetToReturn(assets[0], amounts[0], premiums[0]);
+        return true;
+    }
+
+
+    /**
+     * Runs all the necessary approval functions required for a list of ERC20 tokens.
+     *
+     * @param _tokens    Addresses of the tokens which need approval
+     */
+    function approveTokens(IERC20[] calldata _tokens) external {
+        for (uint256 i = 0; i < _tokens.length; i++) {
+            approveToken(_tokens[i]);
+        }
+    }
+
+    /**
+     * Runs all the necessary approval functions required before issuing
+     * or redeeming a SetToken. This function need to be called only once before the first time
+     * this smart contract is used on any particular SetToken.
+     *
+     * @param _setToken    Address of the SetToken being initialized
+     */
+    function approveSetToken(ISetToken _setToken) isSetToken(_setToken) external {
+        (address longToken,,,) = getLeveragedTokenData(_setToken, 1 ether);
+        approveToken(IERC20(longToken));
+    }
+
+
+
+    /* ============ Internal Functions ============ */
+
+
+
+    /**
+     * Initiates a flashloan call with the correct parameters for issuing set tokens in the callback
+     * Borrows correct amount of long token and and forwards encoded calldata to controll issuance in the callback.
+     *
+     * @param _setToken            Address of the SetToken being initialized
+     * @param _amountSetToken      Amount of the SetToken being initialized
+     * @param _paymentParams        Enum controlling what token the use to pay for issuance
+     */
+    function initiateIssuance(
+        ISetToken _setToken,
+        uint256 _amountSetToken,
+        PaymentToken _paymentToken,
+        bytes memory _paymentParams
+    )
+        isSetToken(_setToken)
+        internal
+        returns (uint256)
+    {
+        (address longToken, uint256 longAmount,,) = getLeveragedTokenData(_setToken, _amountSetToken);
+
+        address[] memory assets = new address[](1);
+        assets[0] = IAToken(longToken).UNDERLYING_ASSET_ADDRESS();
+        uint[] memory amounts =  new uint[](1);
+        amounts[0] = longAmount;
+
+        bytes memory params = abi.encode(_setToken, _amountSetToken, msg.sender, true, _paymentToken, _paymentParams);
```

Comment:
Original question by @ncitron  _Why is _paymentParams encoded as bytes then encoded again here? I think we can just leave it as a uint_